### PR TITLE
CDC Natality - Artifacts

### PR DESCRIPTION
**Pull request by Abilityguy**
Artifacts used for the CDC Natality import.
Source data added using git LFS. Have added a README and tests.

### CODE REVIEW

**Review by ajaits — COMMENTED** [review]
Can we zip files under source_data if it is not used for tests?

**Comment by Abilityguy** [comment]
> Can we zip files under source_data if it is not used for tests?

Done, have zipped source_data.

File: scripts/us_cdc/natality/clean_cdc_data.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+            # Tabs at start of the line are removed
+            cleaned_header = _STARTING_TAB_REGEX.sub('', cleaned_header)
+
+            # Tab delimiter replaced with ,
+            cleaned_header = _TAB_DELIMITER_REGEX.sub(',', cleaned_header)
+
+            w_handle.write(cleaned_header)
+
+        elif not line.startswith('"Total"'):  # Line should not start with total
```
Comment (ajaits) [inline scripts/us_cdc/natality/clean_cdc_data.py:68]: line.startswith("----") should be above this condition?

File: scripts/us_cdc/natality/preprocess.py [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+        statvar_list = []
+        for file_name in os.listdir(_FLAGS.input_path):
+            if file_name.endswith('.txt'):  # Raw data is .txt files
+                file_path = os.path.join(_FLAGS.input_path, file_name)
+                with open(file_path, 'r', encoding='utf-8') as csv_f:
+                    f_reader = csv.DictReader(csv_f,
+                                              delimiter=',',
+                                              quotechar='"')
+                    f_name = file_name[:-4]  # Removing .txt from filename
```
Comment (ajaits) [inline scripts/us_cdc/natality/preprocess.py:211]: Can use os.path.splitext() to remove extension

File: scripts/us_cdc/natality/preprocess_test.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            ])
+
+            with open(csv_path, 'r', encoding='utf-8') as f_result:
+                test_result = f_result.read()
+                expected_csv_path = os.path.join(_SCRIPT_PATH, 'testdata',
+                                                 'expected.csv')
+                with open(expected_csv_path, 'r',
+                          encoding='utf-8') as f_expected:
+                    expected_result = f_expected.read()
+                self.assertEqual(test_result, expected_result)
```
[the remark this answers is not in the document]
Comment (Abilityguy) [reply scripts/us_cdc/natality/preprocess_test.py:47]: Done.

File: scripts/us_cdc/natality/clean_cdc_data.py (hunk 1 above)
Comment (Abilityguy) [reply scripts/us_cdc/natality/clean_cdc_data.py:70]: Done.

File: scripts/us_cdc/natality/preprocess.py (hunk 2 above)
Comment (Abilityguy) [reply scripts/us_cdc/natality/preprocess.py]: Done.

File: scripts/us_cdc/natality/clean_cdc_data.py (hunk 1 above)
Comment (Abilityguy) [reply scripts/us_cdc/natality/clean_cdc_data.py:71]: The ordering of the if elif ladder has changed. The code works as expected. The typical structure of a text file in the source data is as follows:

`Notes ...` (Header line with column names)
`Alabama ...` (Data)
`California ...`
...
`Total ...`
`---` (Start of meta data)

The header block is executed for the first line. The `line.startswith('"Total"')` block is executed for all the data lines. When the meta data line is encountered the script breaks out of the loop and stops writing to the cleaned file.

**Review by ajaits — APPROVED** [review]

**Pull request merged**